docs/extending.rst

```diff
@@ -36,3 +36,43 @@ So it is fairly simple to build your own decorators on top of ``attrs``:
          pass
 
       f = a(b(original_f))
+
+.. _extending_metadata:
+
+Metadata
+--------
+
+If you're the author of a third-party library with ``attrs`` integration, you may want to take advantage of attribute metadata.
+
+Here are some tips for effective use of metadata:
+
+- Try making your metadata keys and values immutable.
+  This keeps the entire ``Attribute`` instances immutable too.
+
+- To avoid metadata key collisions, consider exposing your metadata keys from your modules.::
+
+    from mylib import my_metadata_key
+
+    @attr.s
+    class C(object):
+      x = attr.ib(metadata={my_metadata_key: 1})
+
+  Metadata should be composable, so consider supporting this approach even if you decide implementing your metadata in one of the following ways.
+
+- Expose ``attr.ib`` wrappers for your specific metadata.
+  This is a more graceful approach if your users don't require metadata from other libraries.
+
+  .. doctest::
+
+    >>> my_type_metadata = '__my_type_metadata'
```

Comment:
In my case with app_config I went for `object()`.  It’s a bit ugly in the repr but it’s more fool proof because the keys just *can’t* clash.  Opinions?